src/compiler/transformers/classFields.ts

```diff
@@ -1157,16 +1172,20 @@ namespace ts {
             return pendingExpressions || (pendingExpressions = []);
         }
 
-        function addPrivateIdentifierToEnvironment(node: PrivateClassElementDeclaration) {
+        function addPrivateIdentifierToEnvironment(node: PrivateClassElementDeclaration): Error | void {
             const text = getTextOfPropertyName(node.name) as string;
             const env = getPrivateIdentifierEnvironment();
             const { weakSetName, classConstructor } = env;
             const assignmentExpressions: Expression[] = [];
+
+            const privateName = node.name.escapedText;
+            let potentialDuplicatePrivateName = env.identifiers.has(privateName);

```

Comment:
Why do two lookups in the map (one for the check and one to get `previousInfo` lower), why not just define `previousInfo` and keep the value since we need to do the lookup anyway.